### PR TITLE
支持接收转账消息

### PR DESCRIPTION
**Pull request by dadiyang**
转账消息的MsgType = 49  AppMsgType = 2000

### CODE REVIEW

File: lib/Mojo/Weixin/Message/Handle.pm [hunk 1]
```diff
@@ -295,6 +295,7 @@ sub _parse_sync_data {
             }
             elsif($e->{MsgType} == 49 and $e->{AppMsgType} == 2000){#转账信息
                 $msg->{format} = "payment";
+                $msg->{}
```
Comment (sjdy521) [inline lib/Mojo/Weixin/Message/Handle.pm]: 这里的消息体为空的话，会影响Openwx插件api上报消息结果，我会在此基础上继续完善一下，非常感谢

File: lib/Mojo/Weixin/Message/Handle.pm [hunk 2]
```diff
@@ -438,6 +441,9 @@ sub _parse_sync_data {
                 $self->warn("app message xml parse fail: $@") if $@;
                 $msg->{content} = "[名片]昵称：@{[$msg->{card_name} || '未知']}\n[名片]性别：@{[$msg->{card_sex} || '未知']}\n[名片]位置：@{[$msg->{card_province} || '未知']} @{[$msg->{card_city} || '未知']}\n[名片]头像：@{[$msg->{card_avatar} || '未知']}";
             }
+            elsif($msg->{format} eq "payment"){
+                $msg->{content} = "[转账消息](" . $msg->{content} . ")";
```
Comment (sjdy521) [inline lib/Mojo/Weixin/Message/Handle.pm]: 我没有测试环境，这里最终打印出来的消息长什么样，比如[转账消息](XXXXX) 这里的xxxx会是什么，可以正常获取到吗？

Comment (dadiyang) [reply lib/Mojo/Weixin/Message/Handle.pm]: 这是一个转账的消息实例。content部分收到的是Html内容。<>符号是以\&lt; \&gt;的形式传输的。
转账：
[payment.txt](https://github.com/sjdy521/Mojo-Weixin/files/1481078/payment.txt)
红包：
[redpocket.txt](https://github.com/sjdy521/Mojo-Weixin/files/1481084/redpocket.txt)


{
"BaseResponse": {
"Ret": 0,
"ErrMsg": ""
}
,
"AddMsgCount": 1,
"AddMsgList": [{
"MsgId": "7751000081045396918",
"FromUserName": "@2e339b3dec6f000000000c99f1daeaf67c6a6f62227c6b0c6280b2ba0350000",
"ToUserName": "@000000088dc05b725f9391dce0bd1000",
"MsgType": 49,
"Content": "&lt;msg&gt;<br/>&lt;appmsg appid=\"\" sdkver=\"\"&gt;<br/>&lt;title&gt;&lt;![CDATA[微信转账]]&gt;&lt;/title&gt;<br/>&lt;des&gt;&lt;![CDATA[扬向你转账0.10元。如需收钱，请点此升级至最新版本]]&gt;&lt;/des&gt;<br/>&lt;action&gt;&lt;/action&gt;<br/>&lt;type&gt;2000&lt;/type&gt;<br/>&lt;content&gt;&lt;![CDATA[]]&gt;&lt;/content&gt;<br/>&lt;url&gt;&lt;![CDATA[https://support.weixin.qq.com/cgi-bin/mmsupport-bin/readtemplate?t=page/common_page__upgrade&amp;text=text001&amp;btn_text=btn_text_0]]&gt;&lt;/url&gt;<br/>&lt;thumburl&gt;&lt;![CDATA[https://support.weixin.qq.com/cgi-bin/mmsupport-bin/readtemplate?t=page/common_page__upgrade&amp;text=text001&amp;btn_text=btn_text_0]]&gt;&lt;/thumburl&gt;<br/>&lt;lowurl&gt;&lt;/lowurl&gt;<br/>&lt;extinfo&gt;<br/>&lt;/extinfo&gt;<br/>&lt;wcpayinfo&gt;<br/>&lt;paysubtype&gt;1&lt;/paysubtype&gt;<br/>&lt;feedesc&gt;&lt;![CDATA[￥0.10]]&gt;&lt;/feedesc&gt;<br/>&lt;transcationid&gt;&lt;![CDATA[100005020117111600024321086800000000]]&gt;&lt;/transcationid&gt;<br/>&lt;transferid&gt;&lt;![CDATA[1000050201201711160100300000000]]&gt;&lt;/transferid&gt;<br/>&lt;invalidtime&gt;&lt;![CDATA[1510910500]]&gt;&lt;/invalidtime&gt;<br/>&lt;begintransfertime&gt;&lt;![CDATA[1510818700]]&gt;&lt;/begintransfertime&gt;<br/>&lt;effectivedate&gt;&lt;![CDATA[1]]&gt;&lt;/effectivedate&gt;<br/>&lt;pay_memo&gt;&lt;![CDATA[]]&gt;&lt;/pay_memo&gt;<br/><br/>&lt;/wcpayinfo&gt;<br/>&lt;/appmsg&gt;<br/>&lt;/msg&gt;",
"Status": 3,
"ImgStatus": 1,
"CreateTime": 1510818000,
"VoiceLength": 0,
"PlayLength": 0,
"FileName": "微信转账",
"FileSize": "",
"MediaId": "",
"Url": "https://support.weixin.qq.com/cgi-bin/mmsupport-bin/readtemplate?t=page/common_page__upgrade&amp;text=text001&amp;btn_text=btn_text_0",
"AppMsgType": 2000,
"StatusNotifyCode": 0,
"StatusNotifyUserName": "",
"RecommendInfo": {
"UserName": "",
"NickName": "",
"QQNum": 0,
"Province": "",
"City": "",
"Content": "",
"Signature": "",
"Alias": "",
"Scene": 0,
"VerifyFlag": 0,
"AttrStatus": 0,
"Sex": 0,
"Ticket": "",
"OpCode": 0
}
,
"ForwardFlag": 0,
"AppInfo": {
"AppID": "",
"Type": 0
}
,
"HasProductId": 0,
"Ticket": "",
"ImgHeight": 0,
"ImgWidth": 0,
"SubMsgType": 0,
"NewMsgId": 7751000081045396918,
"OriContent": ""
}
],
"ModContactCount": 0,
"ModContactList": [],
"DelContactCount": 0,
"DelContactList": [],
"ModChatRoomMemberCount": 0,
"ModChatRoomMemberList": [],
"Profile": {
"BitFlag": 0,
"UserName": {
"Buff": ""
}
,
"NickName": {
"Buff": ""
}
,
"BindUin": 0,
"BindEmail": {
"Buff": ""
}
,
"BindMobile": {
"Buff": ""
}
,
"Status": 0,
"Sex": 0,
"PersonalCard": 0,
"Alias": "",
"HeadImgUpdateFlag": 0,
"HeadImgUrl": "",
"Signature": ""
}
,
"ContinueFlag": 0,
"SyncKey": {
"Count": 8,
"List": [{
"Key": 1,
"Val": 644810701
}
,{
"Key": 2,
"Val": 644811091
}
,{
"Key": 3,
"Val": 644811048
}
,{
"Key": 11,
"Val": 644810861
}
,{
"Key": 13,
"Val": 644788661
}
,{
"Key": 201,
"Val": 1510818777
}
,{
"Key": 1000,
"Val": 1510793281
}
,{
"Key": 1001,
"Val": 1510793291
}
]
}
,
"SKey": "",
"SyncCheckKey": {
"Count": 8,
"List": [{
"Key": 1,
"Val": 644810701
}
,{
"Key": 2,
"Val": 644811091
}
,{
"Key": 3,
"Val": 644811041
}
,{
"Key": 11,
"Val": 644810869
}
,{
"Key": 13,
"Val": 644788661
}
,{
"Key": 201,
"Val": 1510818777
}
,{
"Key": 1000,
"Val": 1510793211
}
,{
"Key": 1001,
"Val": 1510793291
}
]
}

}

File: lib/Mojo/Weixin/Message/Handle.pm (hunk 1 above)
Comment (dadiyang) [reply lib/Mojo/Weixin/Message/Handle.pm]: 不好意思，这一行应该是要删掉的。没用过perl语言，所以只是模仿上面的写法拷贝下来改的而已。
我以后多学学perl这门强大的语言。

File: lib/Mojo/Weixin/Message/Handle.pm (hunk 2 above)
Comment (sjdy521) [reply lib/Mojo/Weixin/Message/Handle.pm]: 感谢提供了抓包数据，转账功能已经支持

**Comment by sjdy521** [comment]
感谢提供PR，已经提取了关键代码加入到主分支，现在已经支持接收转账消息

File: lib/Mojo/Weixin/Message/Handle.pm (hunk 2 above)
Comment (dadiyang) [reply lib/Mojo/Weixin/Message/Handle.pm]: 好的。我怎么使用新的版本呢？

**Comment by sjdy521** [comment]
使用方法可以看下FAQ.md

**Pull request closed without merging**